src/algos/swarm.py

```diff
@@ -117,34 +119,34 @@ def single_round(self,self_repr):
     def run_protocol(self):
         start_epochs = self.config.get("start_epochs", 0)
         total_epochs = self.config["epochs"]
+        num_clients = self.config["num_clients"]
+        train_accs = np.zeros((num_clients, total_epochs))
+
         for round in range(start_epochs, total_epochs):
-            #self.log_utils.logging.info("Client waiting for semaphore from {}".format(self.server_node))
-            #print("Client waiting for semaphore from {}".format(self.server_node))
             self.comm_utils.wait_for_signal(src=0, tag=self.tag.START)
-            #print("semaphore received, start local training")
-            # self.log_utils.logging.info("Client received semaphore from {}".format(self.server_node))
-            self.local_train()
-            #self.local_test()
+            train_acc = self.local_train()
+            train_accs[self.node_id-1, round] = train_acc
+            print("Node {} train_acc:{:.4f}".format(self.node_id, train_acc))
+            np.save('./train_accs.npy', train_accs)
+            print(train_accs)
+            self.comm_utils.send_signal(dest=0, data=train_acc, tag=self.tag.FINISH)
+
             self_repr = self.get_representation()
-            # self.log_utils.logging.info("Client {} sending done signal to {}".format(self.node_id, self.server_node))
-            #print("sending signal to node {}".format(self.server_node))
             if self.node_id == 1:
                 repr = self.single_round(self_repr)
-            # self.log_utils.logging.info("Client {} waiting to get new model from {}".format(self.node_id, self.server_node))
-            #get all representation from server
             else:
                 self.comm_utils.send_signal(dest=self.server_node, data=self_repr, tag=self.tag.DONE)
                 print("Node {} waiting signal from node 1".format(self.node_id))
                 repr = self.comm_utils.wait_for_signal(src=self.server_node, tag=self.tag.UPDATES)
                 
-
-            
             self.set_representation(repr)
-            #test updated model 
             acc = self.local_test()
             print("Node {} test_acc:{:.4f}".format(self.node_id, acc))
             self.comm_utils.send_signal(dest=0, data=acc, tag=self.tag.FINISH)
 
+        # Save the train_accs array to a text file
+        np.savetxt('train_accs.txt', train_accs, delimiter=',')
```

Comment:
please create a helper function in log_utils so that it gets stored appropriately.